src/utils/keys.ts

```diff
@@ -9,40 +9,47 @@ export type Keys = {
   ordAddress: string;
   walletPubKey: string;
   ordPubKey: string;
+  walletDerivationPath: string;
+  ordDerivationPath: string;
 };
 
-const getWif = (seedPhrase: string, isOrd?: boolean) => {
+const getWifAndDerivation = (seedPhrase: string, isOrd?: boolean) => {
   const seed = bip39.mnemonicToSeedSync(seedPhrase);
   const masterNode = ExtendedPrivateKey.from_seed(seed);
-
   const derivationPath = `m/44'/236'/${isOrd ? "1" : "0"}'/0/0`;
   const childNode = masterNode.derive_from_path(derivationPath);
   const privateKey = childNode.get_private_key();
   const wif = privateKey.to_wif();
 
-  return wif;
+  return { wif, derivationPath };
 };
 
 export const getKeys = (validMnemonic?: string) => {
+  if (validMnemonic) {
+    const isValid = bip39.validateMnemonic(validMnemonic);
+    if (!isValid) throw new Error("Invalid Mnemonic!");
+  }
   const mnemonic = validMnemonic ?? bip39.generateMnemonic();
-  const walletWif = getWif(mnemonic);
-  const walletPrivKey = PrivateKey.from_wif(walletWif);
+  const walletWifAndDP = getWifAndDerivation(mnemonic);
+  const walletPrivKey = PrivateKey.from_wif(walletWifAndDP.wif);
   const walletPubKey = walletPrivKey.to_public_key();
   const walletAddress = walletPubKey.to_address().to_string();
 
-  const ordWif = getWif(mnemonic, true);
-  const ordPrivKey = PrivateKey.from_wif(ordWif);
+  const ordWifAndDP = getWifAndDerivation(mnemonic, true);
+  const ordPrivKey = PrivateKey.from_wif(ordWifAndDP.wif);
   const ordPubKey = ordPrivKey.to_public_key();
   const ordAddress = ordPubKey.to_address().to_string();
 
   const keys: Keys = {
```

Comment:
What value does it bring to have addresses and pubkeys in the export? I can see how it could provide information for the user, but on import, I assume those fields will be ignored? Also, I think using the 1sat names of `ordPk` and `payPk` would allow for seamless imports of those backups.  Having said that, as long as there is a means to import that format, the export doesn't need to be compatible, so this is fine.

Reply:
Yeah I was curious other's thoughts on including the public keys. I will remove them as I agree it's unnecessary.